src/main/java/org/broadinstitute/hellbender/tools/spark/utils/LargeLongHopscotchSet.java

```diff
@@ -0,0 +1,224 @@
+ * not support dynamic resizing.
+ */
+@DefaultSerializer(LargeLongHopscotchSet.Serializer.class)
+public class LargeLongHopscotchSet implements Serializable, QueryableLongSet {
+
+    private static final long serialVersionUID = 1L;
+    private final List<LongHopscotchSet> sets;
+    private int numSets;
+
+    public LargeLongHopscotchSet(final long maxPartitionBytes, final long numElements) {
+        sets = new ArrayList<>();
+        initialize(maxPartitionBytes, numElements);
+    }
+
+    @SuppressWarnings("unchecked")
+    protected LargeLongHopscotchSet(final Kryo kryo, final Input stream) {
+        final boolean oldReferences = kryo.getReferences();
+        kryo.setReferences(false);
+
+        numSets = stream.readInt();
+        sets = new ArrayList<>(numSets);
+        for (int i = 0; i < numSets; i++) {
+            sets.add(kryo.readObject(stream, LongHopscotchSet.class));
+        }
+
+        kryo.setReferences(oldReferences);
+    }
+
+    private static int longHash(final long entryVal) {
+        return (int) SVUtils.fnvLong64(0x4c79ac6a, entryVal);
+    }
+
+    private void initialize(final long maxPartitionBytes, final long numElements) {
+        Utils.validateArg(maxPartitionBytes > 0, "Max partition size must be greater than 0");
+        Utils.validateArg(numElements > 0, "Number of elements must be greater than 0");
+
+        //Largest partition according to the max bytes per partition constraint
+        final int maxPartitionSize = LongHopscotchSet.getLegalSizeBelow(maxPartitionBytes / LongHopscotchSet.bytesPerBucket);
+        //Truncate numElements to largest int possible
+        final int idealPartitionSize = LongHopscotchSet.getPartitionSize(numElements);
+        //Number of elements per partition
+        final int elementsPerPartitionGuess = Math.min(idealPartitionSize, maxPartitionSize);
+        final int partitions = (int) Math.ceil(((double) numElements) / elementsPerPartitionGuess);
+        final int elementsPerPartition = (int)Math.ceil(numElements / (double)partitions);
+
+        for (int i = 0; i < partitions; i++) {
+            sets.add(new LongHopscotchSet(elementsPerPartition));
+        }
+        numSets = sets.size();
+    }
+
+    protected void serialize(final Kryo kryo, final Output stream) {
+        final boolean oldReferences = kryo.getReferences();
+        kryo.setReferences(false);
+
+        stream.writeInt(sets.size());
+        for (final LongHopscotchSet set : sets) {
+            kryo.writeObject(stream, set);
+        }
+
+        kryo.setReferences(oldReferences);
+    }
+
+    public boolean add(final long entryValue) {
+        final int setIndex = setIndexOf(entryValue);
+        final LongHopscotchSet setRef = sets.get(setIndex);
+        return setRef.add(entryValue);
+    }
+
+    public void addAll(final long[] entryValues) {
+        for (final long val : entryValues) {
+            add(val);
+        }
+    }
+
+    public long size() {
```

Comment:
Probably wouldn't be hard to keep your own total size.  But maybe this is unlikely to be called repetitively.